### ISSUE TITLE
Using selection transform methods lead to warning

### ISSUE THREAD

**Issue by oyeanuj** [ianstormtaylor/slate#597]
Hi @ianstormtaylor, I ran into this warning while using `collapseToStartOfNextBlock` on `onKeyDown`.

```
Warning: An object was passed to a Node method instead of a `key` string. This was previously supported, but is being deprecated because it can have a negative impact on performance. The object in question was: Block {key: "1", nodes: List[1], type: "big-heading"}

warn	@	warn.js:32
key	@	normalize.js:95
getDescendant	@	node.js:535
assertDescendant	@	node.js:115
getNextBlock	@	node.js:852
object.(anonymous function)	@	memoize.js:109
collapseToStartOfNextBlock	@	on-selection.js:137
Transform.(anonymous function)	@	transform.js:134
onKeyDown	@	HeadingPlugin.js:36
```

It doesn't show up in production version of Slate so I wasn't able to reproduce on JSFiddle with the same code. But here is that code anyway, if it helps - https://jsfiddle.net/cxcx81jh/1/

Thank you!

**Comment by oyeanuj** [comment]
Also, seeing this warning in the [Auto-Markdown](http://slatejs.org/#/auto-markdown) example.

**Comment by ianstormtaylor** [comment]
Should be fixed now by #610!
